Formula/vte3.rb

```diff
@@ -16,12 +16,19 @@ class Vte3 < Formula
   depends_on "meson" => :build
   depends_on "ninja" => :build
   depends_on "pkg-config" => :build
+  depends_on "vala" => :build
   depends_on "gettext"
   depends_on "gnutls"
   depends_on "gtk+3"
   depends_on macos: :mojave
   depends_on "pcre2"
-  depends_on "vala"
+
+  on_linux do
+    depends_on "gcc" # for C++17
+    depends_on "systemd"
```

Comment:
Based on the build failure we probably need to add `linux-headers` - it's looking for function which should be defined there.

Reply:
I tried searching for `TIOCGPTPEER` in `linux-headers`, but it isn't available there:
```console
# grep -R TIOCGPT /home/linuxbrew/.linuxbrew/opt/linux-headers/include/*
/home/linuxbrew/.linuxbrew/opt/linux-headers/include/asm-generic/ioctls.h:#define TIOCGPTN	_IOR('T', 0x30, unsigned int) /* Get Pty Number (of pty-mux device) */
/home/linuxbrew/.linuxbrew/opt/linux-headers/include/asm-generic/ioctls.h:#define TIOCGPTLCK	_IOR('T', 0x39, int) /* Get Pty lock state */
```

Based on https://man7.org/linux/man-pages/man4/tty_ioctl.4.html, it mentioned `TIOCGPTPEER` should be available since Linux 4.13

Homebrew's `linux-headers` is v4.4.80.

Reply:
I guess there was a previous attempt to update `linux-headers` to v5 #69081

Alternatively, there was plans to keep this aligned with docker image (is it 5.10.25?):
https://github.com/Homebrew/homebrew-core/pull/64211#issuecomment-723656422

Probably blocked until we update that.

Reply:
We will have to make versioned formula for newer headers.  @sjackman suggested we use version 4.15 and make a formula called `linux-headers@4.15` and also add an alias `linux-headers@4.4` for the current `linux-headers` formula.

Reply:
We will need this for `qemu` as well, so this will probably help with several formulae.

Reply:
Is there a reason we need to keep v4.4 as the main formula rather than using a newer version?

v4.15 is already EOL.
v4.14 (EOL 01/2024) or v4.19 (EOL 12/2024) would probably be better.

Or jumping on the v5 train like v5.4 (EOL 2025)  or 5.10 (EOL 2026).
Though, v5 would probably make things less compatible.

Reply:
I think 4.15 was suggested because that is version that comes with Ubuntu 18.04 LTS, which will be upgrading to after the core merge is done.  Your point about using an LTS version as defined by the kernel upstream (not Ubuntu) is well taken though.  I think @sjackman should really weigh in on this - I don't have a strong opinion either way on whether we should go with Ubuntu LTS or upstream kernel LTS.

Reply:
Okay, I think I see reason.
Ubuntu 16.04 uses 4.4.0 as main package
Ubuntu 18.04 uses 4.15.0 as main package.

Reply:
@danielnachun I see that it makes sense to align to whatever OS version bottles are built on, i.e. Ubuntu's `linux-headers-generic`

We can go with that and maybe swap names/aliases when moving to 18.04 LTS.

Reply:
Yeah that's what we were planning.  Feel free to create a new formula for the 4.15 headers, or let me know if you'd like me to do it.  We can also try to bottle `qemu` after that is merged, assuming its dependencies are bottled as well.

Reply:
You can go ahead.

Reply:
I've submitted the a new formula for `linux-headers@4.15`.  It passed CI fine so as soon as it is approved, merged and bottled, we can rebase this and add it as a dependency.